dora/src/cannon/codegen.rs

```diff
@@ -1101,9 +1101,47 @@ where
     }
 
     fn emit_invoke_direct_void(&mut self, fct_id: FctId, start_reg: Register, num: u32) {
-        assert_eq!(self.bytecode.register_type(start_reg), BytecodeType::Ptr);
+        self.emit_invoke_direct(fct_id, start_reg, num, None);
+    }
+
+    fn emit_invoke_direct_generic(
+        &mut self,
+        dest: Register,
+        fct_id: FctId,
+        start_reg: Register,
+        num: u32,
+    ) {
+        let bytecode_type = self.bytecode.register_type(dest);
+        let offset = self.bytecode.register_offset(dest);
+
+        let reg = self.emit_invoke_direct(fct_id, start_reg, num, Some(bytecode_type));
+
+        self.asm
+            .store_mem(bytecode_type.mode(), Mem::Local(offset), reg);
+    }
+
+    fn emit_invoke_direct(
+        &mut self,
+        fct_id: FctId,
+        start_reg: Register,
+        num: u32,
+        bytecode_type: Option<BytecodeType>,
+    ) -> ExprStore {
         assert!(num > 0);
 
+        let bytecode_type_self = self.bytecode.register_type(start_reg);
+        let offset_self = self.bytecode.register_offset(start_reg);
+        let position = self.bytecode.offset_position(self.current_offset.to_u32());
+        assert_eq!(bytecode_type_self, BytecodeType::Ptr);
+
+        self.asm.load_mem(
+            bytecode_type_self.mode(),
+            REG_RESULT.into(),
+            Mem::Local(offset_self),
+        );
+        self.asm
+            .test_if_nil_bailout(position, REG_RESULT.into(), Trap::NIL);
```

Comment:
AFAICS there is no test for the nil-check. Might be worth to add one.

Reply:
As I plan to activate some existing tests in the near future, I could activate `nil3.dora`. This should be sufficient.